pkg/meta/annotations.go

```diff
@@ -875,3 +880,26 @@ func getResource(annotations map[string]string, annotationName, defValStr string
 func genAnnotationName(prefix, name string) string {
 	return fmt.Sprintf("%s-%s", prefix, name)
 }
+
+// Find all annotation keys containing a certain prefix.
+func findPrefixedAnnotations(annotations map[string]string, prefix string) ([]string, bool) {
+	var matching []string
+	for k := range annotations {
+		if strings.HasPrefix(k, prefix) {
+			matching = append(matching, k)
+		}
+	}
+	return matching, len(matching) > 0
+}
+
+// Generate annotation name for subcluster-specific drain start time
+// eg vertica.com/drain-start-time-sc1
+func GenSubclusterDrainStartAnnotationName(scName string) string {
+	return genAnnotationName(DrainStartSubclusterPrefixAnnotation, scName)
+}
+
+// Find any annotation names with the drain start prefix
+// e.g., "vertica.com/drain-start-time-sc1"
+func FindDrainTimeoutSubclusterAnnotations(annotations map[string]string) ([]string, bool) {
```

Comment:
You could extend this function to also get a suffix (sc name) and return true as soon as there is an annotation with prefix "vertica.com/drain-start-time-" but suffix different from the given suffix. Then you could add a helper in api/helpers.go , owned by VDB, that will call this function.